src/Jellyfin.Networking/AutoDiscoveryHost.cs

```diff
@@ -141,11 +149,10 @@ private async Task RespondToV2Message(IPAddress responderIp, IPEndPoint endpoint
 
         var response = new ServerDiscoveryInfo(localUrl, _appHost.SystemId, _appHost.FriendlyName);
 
-        using var responder = new UdpClient(new IPEndPoint(responderIp, PortNumber));
         try
         {
             _logger.LogDebug("Sending AutoDiscovery response");
-            await responder
+            await udpClient
```

Comment:
Won't work on all platforms. At least macOS will prohibit such usage.

Reply:
Why?

Reply:
No why. macOS prohibits using udpClient bind to broadcast addresses as a responder to send to unicast addresses because the responder IP is too ambiguous in this case. You have to be explicit the responder IP and specify it.

Reply:
I would suggest adding OS specific code then. I don't think going back to 0.0.0.0 is the right choice

Reply:
Why? The interface restriction **never** worked as intended and I personally don't like all the "half-working" implementation that only makes the logic more complicated and actually does nothing because we are still binding to 255.255.255.255 and providing support to respond this address, which inherently binds to all interfaces and does not follow the server interface configuration anyway. OS specific quirks are everywhere:

- You cannot recieve interface scoped broadcasts without binding to 255.255.255.255 first on Linux for some old kernel on Linux
- You cannot prevent Windows to bind to 255.255.255.255

To actually enforce the interface binding, we need to implement a workaround that drop everthing recieved on 255.255.255.255, but this will break almost all current clients.

macOS will work with binding with interface-scoped broadcast addresses, but if we want to keep current behavior and provide support for responding 255.255.255.255, we actually have to bind to 0.0.0.0 to receive broadcasts to 255.255.255.255 on macOS because you cannot bind to 255.255.255.255 along, you have to bind to all addresses to actually receive packet send to "all addresses".

But on all platforms, 255.255.255.255 will listen on all interfaces which already breaks our intended behavior to listen only on configured interfaces, which I don't see any benefit over just listen to all addresses.